lib/Dialect/FIRRTL/Transforms/ExtractInstances.cpp

```diff
@@ -1080,8 +1151,54 @@ void ExtractInstancesPass::createTraceFiles() {
     builder.create<sv::VerbatimOp>(builder.getUnknownLoc(), buffer,
                                    ValueRange{}, builder.getArrayAttr(symbols));
   }
+  if (!classFields.empty()) {
+    addPortsToClass(classFields, extractMetadataClass);
+    // This extract instances metadata class, now needs to be instantiated
+    // inside the SifiveMetadata class. This also updates its signature, so keep
+    // the object of the SifiveMetadata class updated.
+    auto builderOM = mlir::ImplicitLocOpBuilder::atBlockEnd(
+        sifiveMetadataClass->getLoc(), sifiveMetadataClass.getBodyBlock());
+    SmallVector<std::pair<Value, Twine>> classFields = {
+        {builderOM.create<ObjectOp>(
+             extractMetadataClass,
+             builderOM.getStringAttr("extract_instances_metadata")),
+         "extractedInstances_field"}};
+
+    addPortsToClass(classFields, sifiveMetadataClass);
+    auto *node = instanceGraph->lookup(sifiveMetadataClass);
+    assert(node && node->hasOneUse());
+    ObjectOp metadataObj =
+        dyn_cast_or_null<ObjectOp>((*node->usesBegin())->getInstance());
+    assert(metadataObj);
```

Comment:
If it's just going to be asserted to exist with no error message, maybe you can just use `cast` above? Or, do something like `assert(metadataObj && "expected to find a metadataObj")`